pkg/trace/obfuscate/sql.go

```diff
@@ -195,20 +223,24 @@ type tableFinderFilter struct {
 // Filter implements tokenFilter.
 func (f *tableFinderFilter) Filter(token, lastToken TokenKind, buffer []byte) (TokenKind, []byte, error) {
 	switch lastToken {
-	case From:
+	case From, Join:
 		// SELECT ... FROM [tableName]
 		// DELETE FROM [tableName]
+		// ... JOIN [tableName]
 		if r, _ := utf8.DecodeRune(buffer); !unicode.IsLetter(r) {
 			// first character in buffer is not a letter; we might have a nested
 			// query like SELECT * FROM (SELECT ...)
 			break
 		}
 		fallthrough
-	case Update, Into, Join:
+	case Update, Into:
 		// UPDATE [tableName]
 		// INSERT INTO [tableName]
-		// ... JOIN [tableName]
-		f.storeName(string(buffer))
+		token = Table
+
+		if config.HasFeature("table_names") {
```

Comment:
With the risk of becoming annoying, I have to apologise here. I think your previous version was much better. Both your argument about immutability makes sense, and also calling this (and implicitly the underlying `os.Getenv`) on every single token is plain stupid and a waste of resources.

Completely my bad. Can we please change it back?